src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/DisableAzureDiskEncryption.cs

```diff
@@ -286,9 +314,9 @@ public override void ExecuteCmdlet()
                     // | Linux   | Not Specified | N/A                        |
                     // +---------+---------------+----------------------------+
 
-                    if (OperatingSystemTypes.Windows.Equals(currentOSType) &&
-                        !string.IsNullOrEmpty(VolumeType) &&
-                        VolumeType.Equals(AzureDiskEncryptionExtensionContext.VolumeTypeData, StringComparison.InvariantCultureIgnoreCase))
+                    if ((OperatingSystemTypes.Windows.Equals(currentOSType) && parameters.TypeHandlerVersion.Equals(AzureDiskEncryptionExtensionContext.ExtensionSinglePassVersion)) ||
```

Comment:
This will cause UpdateVmEncryptionSettings to be called for Windows VMs + Two pass encryption + Data volume type case. Which shouldn't be.

Reply:
It still skips UpdateVmEncryptionSettings for DualPass-Windows-DataVolume, but adds a skip for SinglePass-Windows and SinglePass-Linux.